core/controllers/base_test.py

```diff
@@ -288,6 +293,25 @@ def test_keys_match_en_qqq(self):
         qqq_key_list = self._extract_keys_from_json_file('qqq.json')
         self.assertEqual(en_key_list, qqq_key_list)
 
+    def test_keys_in_source_code_match_en(self):
+        """Tests that keys in HTML files are present in en.json."""
+        en_key_list = self._extract_keys_from_json_file('en.json')
+        for root, _, files in os.walk(os.path.join(
+                os.getcwd(), self.get_static_asset_filepath())):
+            for filename in files:
+                if filename.endswith('.html'):
+                    html_key_list = self._extract_keys_from_html_file(
+                        os.path.join(root, filename))
+                    self.assertLessEqual(set(html_key_list), set(en_key_list))
+                    if set(html_key_list) > set(en_key_list):
```

Comment:
How do you suggest I do this? Having an assert statement would fail the test as soon as one such key is found, whereas the goal should be to find all the missing keys, that is, let the entire loop run through. If I just log the missing keys, the test doesn't technically "fail" either.